inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cmd/CommandExecutorImpl.java

```diff
@@ -0,0 +1,135 @@
+        }
+        return buffer.toString();
+    }
+
+    private String join(String... strings) {
+        StringBuffer buffer = new StringBuffer();
+        for (String string : strings) {
+            buffer.append(string).append(InlongConstants.BLANK);
+        }
+        return buffer.toString();
+    }
+
+    @Override
+    public CommandResult exec(String cmd) throws Exception {
+        ShellTrackerImpl shellTracker = new ShellTrackerImpl();
+        ShellExecutorImpl shellExecutor = new ShellExecutorImpl(shellTracker);
+        shellExecutor.syncExec("sh", "-c", cmd);
+        String cmdMsg = join("sh", "-c", cmd);
+        LOG.debug("run command : " + cmdMsg);
+        CommandResult commandResult = new CommandResult();
+        commandResult.setCode(shellTracker.getCode());
+        commandResult.setResult(join(shellTracker.getResult()));
+        if (commandResult.getCode() != 0) {
+            throw new Exception("command " + cmdMsg + " exec failed, code = " +
+                    commandResult.getCode() + ", output = " + commandResult.getResult());
+        }
+        LOG.debug(commandResult.getResult());
+        return commandResult;
+    }
+
+    @Override
+    public CommandResult execRemote(AgentClusterNodeRequest clusterNodeRequest, String cmd) throws Exception {
+        String cmdShell = "./conf/exec_cmd.exp";
+        String ip = clusterNodeRequest.getIp();
+        String port = String.valueOf(clusterNodeRequest.getSshPort());
+        String user = clusterNodeRequest.getUsername();
+        String password = clusterNodeRequest.getPassword();
+        String remoteCommandTimeout = "20000";
+
+        cmd = "sh -c \"" + cmd + "\"";
+        String cmdMsg = join(cmdShell, ip, user, password, remoteCommandTimeout, cmd, port);
+        LOG.info("run remote command : " + cmdMsg);
+
+        ShellTrackerImpl shellTracker = new ShellTrackerImpl();
+        ShellExecutorImpl shellExecutor = new ShellExecutorImpl(shellTracker);
+        shellExecutor.syncExec(cmdShell, ip, user, password, remoteCommandTimeout, cmd, port);
+
+        CommandResult commandResult = new CommandResult();
+        commandResult.setCode(shellTracker.getCode());
+        commandResult.setResult(join(shellTracker.getResult()));
+
+        LOG.debug(commandResult.getResult());
+        if (commandResult.getCode() != 0) {
+            throw new Exception(
+                    "remote command " + cmdMsg + " exec failed, code = " + commandResult.getCode() + ", output = "
+                            + commandResult.getResult());
+        }
+        return commandResult;
+    }
+
+    @Override
+    public CommandResult modifyConfig(AgentClusterNodeRequest clusterNodeRequest, Map<String, String> configMap,
+            String confPath)
+            throws Exception {
+        List<String> configList = configMap.entrySet().stream()
+                .map(entry -> "grep " + entry.getKey() + " " + confPath + " && sed -i 's%^" + entry.getKey() + ".*%"
```

Comment:
extract one method for the map operation

Reply:
No need.